code/datums/components/parry.dm

```diff
@@ -0,0 +1,39 @@
+/// Add to a living mob to allow them to "parry" projectiles by clicking on their tile, sending them back at the firer.
+/datum/component/projectile_parry
+	dupe_mode = COMPONENT_DUPE_SOURCES
```

Comment:
after looking at dupe sources its... not the way i envisioned it much. this'll have to be a normal component for now

Reply:
sure thing chief